File: JP/言語非依存の勉強会/アルゴリズム/答え/BubbleSort.php
<?php

$arr = array(5,2,9,6,1);

var_dump(bubble_sort($arr));

function bubble_sort($array)
{
    // 要素数回繰り返し
    for($i = 0; $i < count($array); $i++)
    {
        // 要素数-1回繰り返し
        for($n = 1; $n < count($array); $n++)
        {
            // 隣接要素を比較し大小が逆なら入替える
            if($array[$n-1] > $array[$n])
            {
                $temp = $array[$n];
                $array[$n] = $array[$n-1];
                $array[$n-1] = $temp;
            }
        }
    }
    return $array;
}

File: JP/言語非依存の勉強会/アルゴリズム/答え/BubbleSort.java
public class BubbleSort {
    public static void main(String args[]) {
        int a[] = {5,2,9,6,1};
        // aの内容を表示
        for (int i = 0; i < a.length; i++) {
            System.out.print(a[i] + " ");
        }
        System.out.println();
        // バブルソート
        boolean flag = true; // 交換が発生したかどうか
        while (flag) {
            flag = false;
            // 先頭から順に隣接する逆順要素を交換する
            for (int i = 1; i < a.length; i++) {
                if (a[i-1] > a[i]) {
                    // a[i]とa[i-1]を交換
                    int w = a[i];
                    a[i] = a[i-1];
                    a[i-1] = w;
                    flag = true;
                }
            }
            // aの内容を表示
            for (int i = 0; i < a.length; i++) {
                System.out.print(a[i] + " ");
            }
            System.out.println();
        }
    }
}